Adds initial builder-focal molecule scenario

kushaldas opened the pull request:
## Status

Ready for review

## Description of Changes

Fixes #5446

This will enable creating the builder image sd-builder-focal
from the Dockerfile in this commit. We can then update the next
image_hash in the next commit.


## Testing

- [ ] `make build-debs-focal` command should start building packages using the Focal container image. But, the actual debian package builds will fail. The fixes for the actual packages will come in future PRs.

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

## Checklist

### If you made changes to the server application code:

- [ ] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

### If you made changes to documentation:

- [ ] Doc linting (`make docs-lint`) passed locally

### If you added or updated a code dependency:

Choose one of the following:

- [ ] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
- [ ] I would like someone else to do the diff review

kushaldas committed:
Adds initial builder-focal

This will enable creating the builder image sd-builder-focal
from the Dockerfile in this commit. We can then update the next
image_hash in the next commit.

Commit:
Updates dh-virtualenv focal install

The dh-virtualenv package from sid works well enough. Hopefully upstream
will provide a focal package, but in the meantime, let's install from
sid. Updated the install logic to verify keys via apt, since the
previous iteration was only appropriate for a research spike.

Commit:
Pushes builder image for Ubuntu Focal 20.04

Based on the build logic scenario for Focal, pushing a newly created
image to quay.io/freedomofpress/sd-docker-builder-focal

Also updates a few lingering "xenial" references in the build scenario
for focal.

Commit:
Removes unused xenial build var

Back from the Trusty -> Xenial migration, we still had a var
"securedrop_build_xenial_support" that wasn't actually used anywhere.
Removing it to eliminate confusion prior to the Xenial -> Focal
migration.

conorsch commented:
@kushaldas All set! I've pushed the new focal image based on the logic you've added here. I also took the liberty of appending a few commits. On my machine, I was able to pull the image, confirm it was really Focal, and even build most of the packages—the failing ones are already tracked for follow-up in #5449.

Before you mark as ready-for-review, please review the additional commits to make sure you agree with them, then add some steps to the test plan for the next reviewer!

kushaldas committed:
Fixes #5454 make build-debs-focal

This adds a new build target to build the Debian packages on Focal.
It also includes two variables to be used as ansible override
during future build steps in Ansible.

conorsch commented:
Ran the new makefile target, and i'm getting new Focal packages!

```
 $ tree build/
build/
├── focal
│   ├── securedrop-config-0.1.3+1.6.0~rc1-amd64.deb
│   ├── securedrop-grsec-4.14.188-amd64.deb
│   ├── securedrop-keyring-0.1.4+1.6.0~rc1-amd64.deb
│   ├── securedrop-ossec-agent-3.6.0+1.6.0~rc1-amd64.deb
│   └── securedrop-ossec-server-3.6.0+1.6.0~rc1-amd64.deb
└── xenial
    ├── ossec-agent-3.6.0-amd64.deb
    ├── ossec-server-3.6.0-amd64.deb
    ├── securedrop-app-code_1.6.0~rc1+xenial_amd64.deb
    ├── securedrop-app-code_1.6.0~rc1+xenial.tar.gz
    ├── securedrop-config-0.1.3+1.6.0~rc1-amd64.deb
    ├── securedrop-grsec-4.14.188-amd64.deb
    ├── securedrop-keyring-0.1.4+1.6.0~rc1-amd64.deb
    ├── securedrop-ossec-agent-3.6.0+1.6.0~rc1-amd64.deb
    └── securedrop-ossec-server-3.6.0+1.6.0~rc1-amd64.deb

```

As expected, there are still a few failing packages for focal (viz. `securedrop-app-code`, `ossec-agent`, `ossec-server`), but those follow-ups are well tracked in the epic, so approving for merge.
